src/layout/layout.rs

```diff
@@ -443,19 +443,56 @@ impl Layout {
             .map(|_| Element::constrain(&mut solver, (area_start, area_end)))
             .try_collect()?;
 
-        // ensure there are no gaps between the elements
-        for pair in elements.windows(2) {
-            solver.add_constraint(pair[0].end | EQ(REQUIRED) | pair[1].start)?;
-        }
-
         match layout.flex {
+            Flex::SpaceBetween => {
+                let mut spacers: Vec<Element> = elements
+                    .iter()
+                    .map(|_| Element::constrain(&mut solver, (area_start, area_end)))
+                    .try_collect()?;
+                spacers.pop();
```

Comment:
looks like you're just duplicating a certain number here. can this be something like :
```rust
let spaces = vec![Element::constrain(...) ; elements.len() - 1];

Reply:
This doesn't work unfortunately, because this expands to something like the following:

```rust
let mut spaces = vec![];
let only_one_element_instantiated = Element::constrain(...);
for i in 0..elements.len() - 1 {
  spaces.push(only_one_element_instantiated)
}
```

We need this instead:

```
let mut spaces = vec![];
for i in 0..elements.len() - 1 {
  spaces.push(Element::constrain(...))
}
```

I can probably use `repeat_with` and `take`.

Reply:
https://doc.rust-lang.org/std/vec/struct.Vec.html#method.fill_with perhaps?

Reply:
repeat_with and take seem idiomatic so I'll leave it at that.